src/Framework/ValidationRules/Rules/File.php

```diff
@@ -0,0 +1,110 @@
+<?php
+declare(strict_types=1);
+
+namespace Give\Framework\ValidationRules\Rules;
+
+use Closure;
+use Give\Framework\Http\Types\FileRequest;
+use Give\Vendors\StellarWP\Validation\Contracts\ValidationRule;
+
+/**
+ * @unreleased
+ */
+class File implements ValidationRule
+{
+    /**
+     * The size, in bytes, of the uploaded file
+     *
+     * @var int
+     */
+    protected $maxSize;
+
+    /**
+     * @var string[]
+     */
+    protected $allowedMimeTypes;
+
+    /**
+     * @unreleased
+     */
+    public static function id(): string
+    {
+        return 'file';
+    }
+
+    /**
+     * @unreleased
+     */
+    public function maxSize(int $maxSize): ValidationRule
+    {
+        $this->maxSize = $maxSize;
+
+        return $this;
+    }
+
+    /**
+     * @unreleased
+     */
+    public function getMaxSize(): int
+    {
+        return $this->maxSize;
+    }
+
+    /**
+     * @unreleased
+     */
+    public function allowedMimeTypes(array $allowedMimeTypes): ValidationRule
+    {
+        $this->allowedMimeTypes = $allowedMimeTypes;
+
+        return $this;
+    }
+
+    /**
+     * @unreleased
+     *
+     * @return string[]
+     */
+    public function getAllowedMimeTypes(): array
+    {
+        return $this->allowedMimeTypes ?? [];
+    }
+
+    /**
+     * @unreleased
+     **/
+    public function __invoke($value, Closure $fail, string $key, array $values)
+    {
+        try {
+            $fileRequest = FileRequest::fromArray($value);
```

Comment:
Yes, this is taking the value from the `$_FILES` request during implementation - everything else is coming from `$_REQUEST`.  I think the raw value of a file array still makes sense for a somewhat arbitrary rule. The little `FileRequest` object just makes it easier to deal with for us, but outside of our system (validation library) this could just be interacting with the actual file array value that was sent to the server through $_FILES https://www.php.net/manual/en/reserved.variables.files.php

Reply:
After some discussion, we're going to keep this simple for now and explore a more robust file handling system in the future.